app/validators/questionnaire_schema.py

```diff
@@ -145,6 +146,10 @@ def __init__(self, schema):
 
         self._answers_with_context = {}
 
+    @lru_cache
+    def get_block_ids_by_block_type(self, block_type: str) -> list[str]:
```

Comment:
Naming suggests you are getting back a mapping instead of a list. Change to `get_block_ids_for_block_type`?